src/python/WMCore/JobSplitting/JobFactory.py

```diff
@@ -27,11 +24,11 @@ class JobFactory(WMObject):
     def __init__(self,
                  package='WMCore.DataStructs',
                  subscription=None,
-                 generators=[],
+                 generators=None,
                  limit = 0):
         self.package = package
         self.subscription  = subscription
-        self.generators    = generators
+        self.generators    = generators if generators else []
```

Comment:
That's why I made my comment, because implicitly the default value is still []  
If we default to [] 3 lines below the previous location, then just leave it as it was before :)